src/ngraph/op/non_max_suppression.hpp

```diff
@@ -88,6 +90,7 @@ namespace ngraph
             protected:
                 BoxEncodingType m_box_encoding = BoxEncodingType::CORNER;
                 bool m_sort_result_descending = true;
+                std::string m_output_type = "i64";
```

Comment:
LGTM overall but I'd suggest to adjust the implementation in the following way:
- switch the member from `string` to `element::Type`
- implement 2 constructors, one that takes an enum and one that takes a string
- in the second constructor use a helper that converts a string to an enum https://github.com/NervanaSystems/ngraph/pull/4432/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.hpp#diff-b358c2a0e091411efef86dea664fb6cdR37
- this helper automatically checks if the string's value passed to the constructor is allowed

Reply:
Switch from `string` to `element::Type` is done.  

I did not implement the 2nd constructor which converts from `string` to `enum` as you suggested.  I am following the `ShapeOf` example mentioned by @diyessi above which has no such constructor.  Let me know if you think it is required.

Reply:
This double constructor approach(one with steing and one with enum) comes from the times when we were working on opset1. Now with the visitor API it might not be needed anymore. @diyessi @lazarevevgeny any thoughts?

Reply:
Yes, with the visitor API the constructor with string attribute does not look like very important. I don't have strong preference in this case.